lib/collection/src/update_handler.rs

```diff
@@ -352,4 +344,34 @@ impl UpdateHandler {
             .send(OptimizerSignal::Stop)
             .unwrap_or_else(|_| debug!("Optimizer already stopped"));
     }
+
+    async fn flush_worker(
+        segments: LockedSegmentHolder,
+        wal: Arc<Mutex<SerdeWal<CollectionUpdateOperations>>>,
+        flush_interval_sec: u64,
+        mut stop_receiver: oneshot::Receiver<()>,
+    ) {
+        loop {
+            // Stop flush worker on signal or if sender was dropped
+            // Even if timer did not finish
+            tokio::select! {
+                _ = tokio::time::sleep(Duration::from_secs(flush_interval_sec)) => {},
+                _ = &mut stop_receiver => {
+                    debug!("Stopping flush worker.");
+                    return;
+                }
+            };
```

Comment:
Does this mean, that if we have `flush_interval_sec = 10`, we would need to wait for ~5sec in case we want to remove collection?

Reply:
`select` here is exactly to avoid this situation - it stops when the first future is ready, ignoring the second future.

Reply:
I added a comment there on top - even if it is waiting on timer now, it will react to the signal and return

Reply:
So there should be no additional wait time on collection removal or parameter update

Reply:
sorry, missed the comment